src/SignalR.Orleans/Clients/ClientGrain.cs

```diff
@@ -62,7 +63,7 @@ public override async Task OnActivateAsync()
                 subscriptionTasks[i] = subscription.ResumeAsync((serverId, _) => OnDisconnect("server-disconnected"));
             }
 
-            await Task.WhenAll(subscriptionTasks);
+            await Task.WhenAll(subscriptionTasks.Where(task => task != null));
```

Comment:
This should probably be a slice instead: ArrayPool<T>.Rent returns an array which is at *least* the requested length, so the non-null items will always be at the beginning and the null items will always be at the end.

However, given that we need to trim the array and Task.WhenAll will convert whatever you pass it into an array (or is it a list?), it would be cheaper just to declare subscriptionTasks as a new List with no pooling.